File: EatNow admin/Responsive-Admin-Dashboard-master/lib/data/model/product.dart
// model/product.dart
class ProductModel {
  int? id;
  String name;
  double price;
  String img;
  String desc;
  int catid;

  ProductModel({
    this.id,
    required this.name,
    required this.price,
    required this.img,
    required this.desc,
    required this.catid,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'price': price,
      'img': img,
      'desc': desc,
      'catid': catid,
    };
  }

  static ProductModel fromMap(Map<String, dynamic> map) {
    return ProductModel(
      id: map['id'],
      name: map['name'],
      price: map['price'].toDouble(),
      img: map['img'],
      desc: map['desc'],
      catid: map['catid'],
    );
  }
}

File: Service_API (2)/Service_API/models/product.js
const mongoose = require("mongoose");

const ProductSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        unique: true
    },
    price: {
        type: Number,
        requried: true
    },
    quantity: {
        type: Number,
        required: true,
    },
    img: {
        type: String,
        required: true
    },
    Size: {
        type: String,
        required: true

    },
    category_id: {
        type: mongoose.SchemaTypes.ObjectId,
        ref: "Category"
    }
})

const Product = mongoose.model("Product", ProductSchema);

module.exports = Product;